### PR TITLE
feat(many): rewrite color system

### PR DESCRIPTION
**Pull request by HerrTopi**
BREAKING CHANGE:
Breaks color overrides in certain cases

### CODE REVIEW

File: cypress/component/DateInput.cy.tsx [hunk 1]
```diff
@@ -95,7 +95,7 @@ describe('<DateInput/>', () => {
         cy.get('span[class$="-calendarDay__day"]').should(
           'have.css',
           'background-color',
-          'rgb(11, 135, 75)'
+          'rgb(16, 135, 81)'
```
Comment (balzss) [inline cypress/component/DateInput.cy.tsx:98]: might be better to just delete this test

File: cypress/component/Select.cy.tsx [hunk 2]
```diff
@@ -77,8 +77,8 @@ describe('<Select/>', () => {
           return window.getComputedStyle(icon).fill
         })
 
-        expect(iconStyles[0]).to.equal('rgb(252, 94, 19)')
-        expect(iconStyles[1]).to.equal('rgb(3, 116, 181)')
+        expect(iconStyles[0]).to.equal('rgb(203, 78, 21)')
+        expect(iconStyles[1]).to.equal('rgb(17, 124, 186)')
```
Comment (balzss) [inline cypress/component/Select.cy.tsx:81]: and these two

File: packages/__docs__/src/Theme/index.tsx [hunk 3]
```diff
@@ -181,16 +179,24 @@ class Theme extends Component<ThemeProps> {
...
       subSections.push(<ThemeColors colors={baseColors} />)
     }
 
     Object.keys(newData).forEach((key) => {
+      //primitives are the color palette above
+      if (key === 'primitives') {
+        return
+      }
+
+      // if (key === 'contrasts') {
```
Comment (balzss) [inline packages/__docs__/src/Theme/index.tsx:196]: why is this commented out? maybe delete it if we won't need it now or add a TODO comment if it's temporary

**Review by balzss — CHANGES_REQUESTED** [review]

**Review by git-nandor — APPROVED** [review]

File: docs/guides/themes.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+</InstUISettingsProvider>
+```
+
+## Colors
+
+InstUI uses a multi layered color system. These layers are
+
+- Primitives
+- Contrasts
+- UI (colors)
```
Comment (matyasf) [inline docs/guides/themes.md:46]: add a bit explanation here too, e.g.

- Primitives: Base colors
- Contrasts: Theme building block, a color that takes up the value of a `primitive` based on the theme
- UI: Theme-specific color of a UI element like link text

File: docs/guides/themes.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+### Primitives
+
+InstUI has a cross-theme (meaning they are the same for every theme) color palette, the `primitives`:
```
Comment (matyasf) [inline docs/guides/themes.md:50]: rewrite to something like: 
InstUI 's color palette is the same for every theme. The numbers in the color names denote the contrast ratio to XX

File: docs/guides/themes.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+```jsx
+---
+type: example
+---
+<ThemeColors colors={canvas.colors.primitives}></ThemeColors>
+```
+
+Everythig is built using these colors. Ideally, our users should never interact directly with this palette, displayed only for reference.
```
Comment (matyasf) [inline docs/guides/themes.md:59]: Every theme in InstUI is built using these colors. Ideally users should never interact directly with this palette, here its only displayed only for reference.

File: docs/guides/themes.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+Primitives are exposed by the theme object. E.g.: `canvas.primitives.green45`.
+
+### Contrasts
+
+`contrasts` are built from the `primitives`. They are theme specific, but they give hints on how they are used in normal and high contrast themes.
+
+For example, there is the `grey1214` contrast. It has a 4 digit last part, which means that it uses the`grey12` primitive for `canvas` and the`grey14` primitive for `canvas-high-contrast`.
+
+Contrasts are used internally by instUI and they are also the building blocks of the `UI` colors (see below). Ideally, our users should rarely or never interact directly with this abstraction, only in special cases, involving theme overrides or making components "intsUI-style".
```
Comment (matyasf) [inline docs/guides/themes.md:69]: typo: instUI-style

File: packages/emotion/src/__tests__/useTheme.test.tsx [hunk 8]
```diff
@@ -100,78 +100,11 @@ describe('useTheme hook', () => {
...
-            <ExampleComponent callback={callback}></ExampleComponent>
-          </InstUISettingsProvider>
-        </InstUISettingsProvider>
-      )
-
-      expect(callback).to.have.been.calledWithMatch({
-        ...canvasHighContrast,
-        colors: { brand: 'red' }
-      })
-    })
```
Comment (matyasf) [inline packages/emotion/src/__tests__/useTheme.test.tsx:169]: please do not delete these, here we are not testing for something that can be done by Chromatic, but the business logic of the theming engine

File: packages/emotion/src/__tests__/useTheme.test.tsx [hunk 9]
```diff
@@ -185,112 +118,11 @@ describe('useTheme hook', () => {
       expect(callback).to.have.been.calledWith(canvas)
     })
   })
-  describe('with InstUISettingsProvider and global theme', () => {
```
Comment (matyasf) [inline packages/emotion/src/__tests__/useTheme.test.tsx:188]: please do not delete this, here we are not testing for something that can be done by Chromatic, but the business logic of the theming engine

**Review by matyasf — CHANGES_REQUESTED** [review]

**Review by matyasf — APPROVED** [review]
Great work! I've checked all components manually and they look OK and I could not find any bugs.

**Pull request merged**